src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -237,14 +237,42 @@ protected Task Stop()
         {
             if (helpParams == null) { helpParams = "get-help"; }
 
-            var psCommand = new PSCommand();
-            psCommand.AddCommand("Get-Help");
-            psCommand.AddArgument(helpParams);
-            psCommand.AddParameter("Online");
-
-            await editorSession.PowerShellContext.ExecuteCommand<object>(psCommand);
+            PSCommand psCommand = new PSCommand();
+            // Check if it's an actual command first. This returns near instant. Get-Help takes a good 5 seconds to return when not a command.
+            psCommand.AddCommand("Get-Command").AddArgument(helpParams);
+            IEnumerable<object> isCommand = await editorSession.PowerShellContext.ExecuteCommand<object>(psCommand);
+            if (isCommand != null && isCommand.Count() > 0){
+                psCommand = new PSCommand();
+                psCommand.AddCommand("Get-Help");
+                psCommand.AddArgument(helpParams);
+                psCommand.AddCommand("Select-Object").AddArgument("RelatedLinks");
+                IEnumerable<object> relatedLinks = await editorSession.PowerShellContext.ExecuteCommand<object>(psCommand);
+                psCommand = new PSCommand();
+                psCommand.AddCommand("Get-Help");
+                psCommand.AddArgument(helpParams);
+                bool sendOutput = false;
+                // Check if the first returned item converted to string is an empty object.
+                // From experience this is the return when there are no RelatedLinks.
+                if (relatedLinks != null && relatedLinks.First().ToString() != "@{RelatedLinks=}")
```

Comment:
another option (which I'm liking better) is to convert 95% of this logic into a PowerShell script and just use `AddScript`. Especially since this function doesn't return anything, this would be a good time to do it.

Reply:
Funny, I asked about the related links in slack and was told that there would never be a related link without a online version. Clearly this is false, so this needs to be fixed. I like the idea of the powershell script instead. I might even still have one in the commit history (or I might have removed it...)

Reply:
Drilling down a PSObject is kind of obtuse in C#, but it's still probably better to keep it in C#.  I'd just make it a static helper method like this

```csharp
internal static bool TryGetHelpUri(PSObject helpInfo, out string helpUri)
{
    PSObject links = helpInfo.Properties['relatedLinks']?.Value as PSObject;
    if (links == null)
    {
        return false;
    }

    PSObject[] navigation = links.Properties['navigationLink']?.Value as PSObject[];
    if (navigation == null)
    {
        return false;
    }

    helpUri = navigation.FirstOrDefault()?.Properties['uri']?.Value as string;
    return !string.IsNullOrWhiteSpace(helpUri);
}
```

```csharp
PSObject helpInfo = MethodThatInvokesGetHelp();
if (TryGetHelpUri(helpInfo, out string helpUri)
{
    // use help uri
}
```

Reply:
Yeah that's better. 👌